CHANGELOG.md

```diff
@@ -16,8 +16,13 @@
 ### ⬆️ Improved
 
 ### ✅ Added
+- Added `UploadedFile` which represents an uploaded file. It contains the url to the file under the property `file`, and a thumbnail of the file under the property `thumbUrl`. Thumbnails are usually returned when uploading a video file. [#4058](https://github.com/GetStream/stream-chat-android/pull/4058)
 
 ### ⚠️ Changed
+- `ChatClient.sendFile` now returns `UploadedFile` instead of `String`. `UploadedFile.file` is the equivalent of the previous return value. If you do not need the other parts of `UploadedFile`, you can use `.map { it.file }` to mitigate the breaking change. [#4058](https://github.com/GetStream/stream-chat-android/pull/4058)
```

Comment:
Now that it is going to be a breaking change in our `FileUploader`, Could we change the return time of our `FileUploader::sendImage` method? Returning something like `UploadedImage`. It can help us in the future to add extra fields to this class without breaking changes (For example, image size, mime-type, or whatever relevant info our backend wants to add)

Reply:
On the other hand, are `FileUploader::deleteFile` right? Is backend deleting thumbnails when a video file is deleted?
Should we change this method signature as well?
Maybe backend could include some `id` to the `UploadedFile` entity to be sure all related info to this file is removed

Reply:
Added the breaking change warnings.

As per Jc's suggestion. According to the open API `ImageUploadResponse` contains identical fields as `FileUploadResponse` (`duration`, `file`, `thumb_url`) + an additional field of `image_size` so it might be prudent to add this if we're already breaking APIs as Jc suggested.

Should note two things here:

1. I did not add the (upload) `duration` field to `UploadedFile` because it didnot seem lime something particularly useful to us, but of course I can if you think it is.
2. While `thumb_url` exists as a field in `ImageUploadResponse`, it never appeared while monitoring network traffic and indeed BE confirmed that it is not sent. When uploading images I only ever got back the file itself and the duration.

Reply:
Unsure about your second message Jc, may be best to ask the BE folks.

Reply:
Added `UploadedImage`.

Decided on no action when it comes to `FileUploader::deleteFile` as BE confirmed associated resources are cleaned up after cache becomes stale (2 weeks).